spec/lib/bank/checking-account_spec.rb

```diff
@@ -0,0 +1,138 @@
+describe Bank::CheckingAccount do
+  it "creates a new instance" do
+    expect(Bank::CheckingAccount).to respond_to(:new)
+  end
+
+  it "inherits from Bank::Account" do
+    expect(Bank::CheckingAccount.superclass).to eq(Bank::Account)
+  end
+
+  let(:account) { Bank::CheckingAccount.new("Checking", 1000) }
+
+  describe "#withdraw(amount)" do
+    it "withdraws amount from balance" do
+      account.withdraw(100)
+      expect(account.balance).to eq(900)
+    end
+
+    # from an ATM transaction
+
+    # how can I check for the $1 independently?
+    it "also withdraws $1 fee and returns balance" do
+      expect(account.withdraw(100)).to eq(899)
```

Comment:
So one way you can write this test is like this:
```
old_balance = account.balance
fee = 1 #If you had a reader for fee, it would help here
withdraw_amount = 100
new_balance = old_balance - withdraw_amount - fee

expect(account.withdraw(withdraw_amount)).to eq(new_balance)
expect(account.balance).to eq(new_balance)
```